lib/reportDisables.mjs

```diff
@@ -1,58 +1,43 @@
-/** @typedef {import('stylelint').DisableReportRange} DisabledRange */
-/** @typedef {import('stylelint').LintResult} StylelintResult */
-/** @typedef {import('stylelint').ConfigRuleSettings<any, Object>} StylelintConfigRuleSettings */
+import reportCommentProblem from './utils/reportCommentProblem.mjs';
 
 /**
  * Returns a report describing which `results` (if any) contain disabled ranges
  * for rules that disallow disables via `reportDisables: true`.
  *
- * @param {StylelintResult[]} results
+ * @param {import('stylelint').PostcssResult} postcssResult
+ * @returns {void}
  */
-export default function reportDisables(results) {
-	for (const result of results) {
-		// File with `CssSyntaxError` don't have `_postcssResult`s.
-		if (!result._postcssResult) {
-			continue;
-		}
-
-		const rangeData = result._postcssResult.stylelint.disabledRanges;
+export default function reportDisables(postcssResult) {
+	const rangeData = postcssResult.stylelint.disabledRanges;
+	const configRules = postcssResult.stylelint.config?.rules;
 
-		if (!rangeData) continue;
+	if (!configRules) return;
 
-		const config = result._postcssResult.stylelint.config;
+	// If no rules actually disallow disables, don't bother looking for ranges
+	// that correspond to disabled rules.
+	if (!Object.values(configRules).some((rule) => reportDisablesForRule(rule))) {
+		return;
+	}
 
-		if (!config || !config.rules) continue;
+	for (const [rule, ranges] of Object.entries(rangeData)) {
+		for (const range of ranges) {
+			if (!configRules[rule]) continue;
 
-		// If no rules actually disallow disables, don't bother looking for ranges
-		// that correspond to disabled rules.
-		if (!Object.values(config.rules).some((rule) => reportDisablesForRule(rule))) {
-			continue;
-		}
+			if (!reportDisablesForRule(configRules[rule])) continue;
 
-		for (const [rule, ranges] of Object.entries(rangeData)) {
-			for (const range of ranges) {
-				if (!reportDisablesForRule(config.rules[rule] || [])) continue;
-
-				// If the comment doesn't have a location, we can't report a useful error.
-				// In practice we expect all comments to have locations, though.
-				if (!range.comment.source || !range.comment.source.start) continue;
```

Comment:
question

Do all comments have "locations"?

Reply:
Do you mean the comments "// If the comment doesn't have a location, we can't report..."? If so, yes.